In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import os

In [2]:
os.listdir()

['.ipynb_checkpoints',
 'ex1data1.txt',
 'img',
 'Linear Regression Assignment.ipynb',
 'Linear Regression From Scratch.ipynb']

In [3]:
data = np.loadtxt('ex1data1.txt',delimiter=",")

In [4]:
data = pd.DataFrame(data)

In [5]:
data.columns=["Population of city in 10,000","Profit in $10,000"]

In [6]:
data.head()

Population of city in 10,000  Profit in $10,000
0                        6.1101            17.5920
1                        5.5277             9.1302
2                        8.5186            13.6620
3                        7.0032            11.8540
4                        5.8598             6.8233

In [7]:
X  = data['Population of city in 10,000'].values.reshape(-1,1)
y = data['Profit in $10,000'].values

In [8]:
from sklearn.model_selection import train_test_split

In [9]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.30, random_state=0)

In [10]:
from sklearn.linear_model import LinearRegression

In [11]:
Regressor = LinearRegression()

In [12]:
Regressor.fit(X_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [13]:
predict = Regressor.predict(X_test)

In [15]:
##Using the trained model to predict the profit for a population of 5.5
pop = np.array([5.5]).reshape(-1,1)

In [17]:
Regressor.predict(pop)

array([2.73037556])

From above the predicted profit is `2.73` which is closer to the practised trained algorithm result of `2.69`